controller_manager:
  ros__parameters:
    update_rate: 100 # Hz

    neural_controller:
      type: neural_controller/NeuralController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

neural_controller:
  ros__parameters:
    repeat_action: 8

    joint_names:
      [
        "leg_front_r_1",
        "leg_front_r_2",
        "leg_front_r_3",
        "leg_front_l_1",
        "leg_front_l_2",
        "leg_front_l_3",
        "leg_back_r_1",
        "leg_back_r_2",
        "leg_back_r_3",
        "leg_back_l_1",
        "leg_back_l_2",
        "leg_back_l_3",
      ]

    default_joint_pos:
      [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    kps: [5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0]
    kds: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]

    init_kps: [5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0]
    init_kds: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]

    init_duration: 2.0
    fade_in_duration: 2.0

    action_types:
      [
        "position",
        "position",
        "position",
        "position",
        "position",
        "position",
        "position",
        "position",
        "position",
        "position",
        "position",
        "position",
      ]
    action_scales: [0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3, 0.3]

    action_limit: 1.5

    max_body_angle: 1.5

    model_path: "/home/pi/policy.json"

teleop_twist_joy_node:
  ros__parameters:
    axis_linear:
      x: 1
      y: 0
    scale_linear:
      x: 0.75
      y: 0.5

    axis_angular:
      yaw: 3
    scale_angular:
      yaw: 1.5

    require_enable_button: false

joy_node:
  ros__parameters:
    dev: "/dev/input/js0"
    autorepeat_rate: 50.0

joint_state_broadcaster:
  ros__parameters:
    use_local_topics: false
